### DISCUSSION TITLE
HTML Custom Templates - File "unknown"

### DISCUSSION THREAD

**Discussion by BenWunderlich** [jeertmans/manim-slides#397] [Q&A] [answered]
I am working on a project that requires manim slides to be converted to html. I am able to do this with the `convert` command. The issue occurs when I try to use/create a custom reveal.js template. According to this [documentation](https://manim-slides.eertmans.be/latest/reference/customize_html.html), I can use `--use-template` to use my own template. When I try to do this with the template from the [here](https://manim-slides.eertmans.be/latest/reference/customize_html.html#full-code), I receive syntax errors like `  File "<unknown>", line 63, in template, jinja2.exceptions.TemplateSyntaxError: tag name expected`. I am not familiar with jinja so I am not sure if the issue is coming from the actual provided template or my understanding on how to use custom templates. Any ideas would be much appreciated!

**Comment by jeertmans** [comment]
Hello @BenWunderlich, I copy/pasted the code (using the `Copy to clipboard` button on the top right) to a blank file, and I could use it with success (i.e., it renders like in the example). You can also download this file from: https://raw.githubusercontent.com/jeertmans/manim-slides/main/docs/source/_static/template.html.

If that does not work, can you please detail a bit your workflow?

**Reply by BenWunderlich** [reply]
I created a new project from scratch using the basicExample code in the documentation and am still having issues. This is specifically the command I am using to convert after rendering `manim-slides convert BasicExample output.html --use-
template template.html`. When I open the template.html file in VSCode, I do get error lines immediately, so I am not sure if I am missing a specific dependency. 

<img width="1440" alt="Screenshot 2024-04-03 at 8 48 46 AM" src="https://github.com/jeertmans/manim-slides/assets/46753167/c53e3d7d-6aa5-4c11-ad67-5b99d9aad500">

Thanks

**Reply by jeertmans** [reply]
The errors line in VSCode are not relevant, because it tries to parser pure HTML, but it is a Jinja2 HTML template.

There is an extra space in your command, maybe this is the issue?

**Reply by BenWunderlich** [reply]
I think the extra space just got copied over by accident. In general, I have been doing it without the extra space. Im gonna keep messing around with the template file and see if I can figure something out. Thanks!

**Reply by jeertmans** [reply]
Can you post a full error trace back here?

**Reply by BenWunderlich** [reply]
Turns out VScode was formatting the html weirdly which is what was causing the error. It now works. Sorry for the inconvenience.

**Reply by jeertmans** [reply]
No problem, can you document a bit your fix? :)

**Reply by BenWunderlich** [reply, marked as answer]
Yes. I found that VSCode default HTML Formatter will break the template's HTML when you save. I found that turning off HTML Formatter in settings for VSCode fixed the issue.

**Reply by jeertmans** [reply]
Thanks!
